### DISCUSSION TITLE
General instability

### DISCUSSION THREAD

**Discussion by lagerratrobe** [skycocker/chromebrew#7035] [General]
Hi All,

I am a very longtime user of Chromebrew and a huge fan of it.  However, my most recent install of it on a freshly powerwashed Google Chromebook has not been a good experience.  It started with an attempt to get GDAL working and has pretty much gone downhill from there.  What I'm seeing that is most troubling is that packages will simply not build correctly from source right now.  In other words, take a package.rb file, download the source and try to build it using the configure options listed.  I've hit all of the following issues:

* glibc won't build from source, fails with some non-specific error, "Target 'all' not remade because of errors."
* iconv as installed out of the box doesn't properly support Latin1 conversions. 
* Building libiconv solves the Latin1 support problem, but then r.rb can't build because its missing libiconv references (likely glibc related)
* GDAL won't compile from source because of a problem in the `std::numeric_limits` declaration in the code which has to be hacked by adding an include into stdlib.h for "limits" when using C++

Things just feel... unstable at the moment.  Is this something that any of the more senior users of Chromebrew have also been seeing?

I'm continuing to hack away at this, but I'm beginning to suspect that I'm in a dependency-hell loop that I'm not going to be able to get out of.  I want to work towards solving the problem though, so what suggestions do you have to get at the root of these issues?

**Comment by satmandu** [comment]
Iconv issues may be linked to the version of glibc you have on your system.

Newer glibc doesn't required libiconv, which is a package primarily for use on non-glibc systems.

We have moved most packages away from requiring libiconv for over a year.

However, our system may include a version of glibc old enough to have iconv issues.

We do not however replace your system glibc with a newer version, as Chromebrew, unlike crouton, does not use a chroot.

You will have to provide more info about your system as per the issue template to get more support though.
